src/Types/InputOutput/LeafType.hack

```diff
@@ -2,8 +2,10 @@ namespace Slack\GraphQL\Types;
 
 use namespace Slack\GraphQL;
 
-abstract class LeafOutputType extends NamedOutputType {
-    abstract const type TCoerced as mixed = this::THackType;
+abstract class LeafType extends NamedType {
+    use TNonNullableType;
+    use TNamedInputType;
+    use TNamedOutputType;
```

Comment:
It's pretty confusing that `coerce` is used to coerce from a Hack type to a GQL type, but `coerceValue` and `coerceNonVariableNode` are used to coerce from GQL types to Hack types. Can we maybe rename one side of these? Maybe `coerce` could be `serializeValue` or something like that?